File: Kadane's Algorithm/KadaneAlgorithm.py
arr = [-2,-3,4,-1,-2,1,5,-3]
local_maxima = arr[0]
global_maxima = arr[0]
for i in range(1,len(arr)):
    local_maxima = max(arr[i], arr[i] + local_maxima)
    global_maxima = max(local_maxima,global_maxima)
    
print("The Largest Subarray Sum is",global_maxima)

File: Kadane's Algorithm/KadaneAlgorithm.java
import java.util.*;
public class KadaneAlgorithm {
    public static void main(String[] args) {
        int[] arr = {-2,3,2,-1};
        int local_maxima = arr[0];
        int global_maxima = arr[0];

        for(int i=1;i<arr.length;i++) {
            local_maxima = Math.max(arr[i], arr[i] + local_maxima);
            global_maxima = Math.max(local_maxima,global_maxima);
        }

        System.out.println("The Largest Subarray Sum is "+global_maxima);
    }
}

// O(N)